server: TestAdminAPITableStats failed [priority range ID already set]

server.TestAdminAPITableStats [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4289549&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4289549&tab=artifacts#/) on master @ [5c0d5132ade6d2e30a9d414163851272750d8afa](https://github.com/cockroachdb/cockroach/commits/5c0d5132ade6d2e30a9d414163851272750d8afa):

Fatal error:

```
panic: priority range ID already set: old=2, new=44 [recovered]
	panic: priority range ID already set: old=2, new=44

```
Stack: 

```
goroutine 270411 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc002cac360, {0x505f438, 0xc001fccae0})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:251 +0x94
panic({0x389e8c0, 0xc00a4b5260})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc00252bf50, 0x2c)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/scheduler.go:125 +0xb9
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc002a944e0, 0xb)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/scheduler.go:240 +0x58
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc00407aa80, {0x505f438, 0xc006931ce0}, 0xc0079433b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_init.go:385 +0x745
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).loadRaftMuLockedReplicaMuLocked(0xc00407aa80, 0xc0079433b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_init.go:212 +0x635
github.com/cockroachdb/cockroach/pkg/kv/kvserver.prepareRightReplicaForSplit({0x505f438, 0xc006931b30}, 0xc007943340, 0xc001eff500)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_split.go:226 +0x19e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.splitPostApply({0x505f438, 0xc006931b30}, {0x0, 0x16d04231cd60f61b, 0x0, 0xe1, 0x90, 0x3, 0x6f, 0x3, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_split.go:145 +0x4b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleSplitResult(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_application_result.go:247
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*replicaStateMachine).handleNonTrivialReplicatedEvalResult(0xc001eff608, {0x505f438, 0xc006931b30}, 0xc001623040)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_application_state_machine.go:1272 +0x46c
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*replicaStateMachine).ApplySideEffects(0xc001eff608, {0x505f438, 0xc006931b30}, {0x50c0508, 0xc001623008})
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_application_state_machine.go:1154 +0x637
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.mapCheckedCmdIter({0x7f36bff819e0, 0xc001eff878}, 0xc00252d320)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/cmd.go:206 +0x158
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*Task).applyOneBatch(0xc00252d8a8, {0x505f438, 0xc0054e4150}, {0x508f328, 0xc001eff818})
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:291 +0x205
github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply.(*Task).ApplyCommittedEntries(0xc00252d8a8, {0x505f438, 0xc0054e4150})
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/apply/task.go:247 +0x9a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReadyRaftMuLocked(0xc001eff500, {0x505f438, 0xc0054e4150}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_raft.go:890 +0x161e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).handleRaftReady(0xc00252de70, {0x505f438, 0xc0054e4150}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_raft.go:531 +0x15b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).processReady(0xc004e9a900, 0x505f438)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_raft.go:507 +0xd7
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).worker(0xc002a944e0, {0x505f438, 0xc001fccae0})
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/scheduler.go:305 +0x25d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:494 +0x16f
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:483 +0x445
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestAdminAPITableStats
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAdminAPITableStats686845435
    test_log_scope.go:80: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPITableStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-12886

Not too hard to repro using gceworker:

```
499 runs so far, 0 failures, over 6m55s
505 runs so far, 0 failures, over 7m0s
511 runs so far, 0 failures, over 7m5s
517 runs so far, 0 failures, over 7m10s
524 runs so far, 0 failures, over 7m15s
530 runs so far, 0 failures, over 7m20s
536 runs so far, 0 failures, over 7m25s
543 runs so far, 0 failures, over 7m30s

=== RUN   TestAdminAPITableStats
    test_log_scope.go:79: test logs captured to: /tmp/logTestAdminAPITableStats2045598821
    test_log_scope.go:80: use -show-logs to present logs inline
panic: priority range ID already set: old=2, new=45 [recovered]
        panic: priority range ID already set: old=2, new=45

goroutine 505 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc000694870, {0x506a918, 0xc002f9c480})
        /home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:251 +0x94
panic({0x3876f80, 0xc0044099a0})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*rangeIDQueue).SetPriorityID(0xc0013f1f20, 0x2d)
        /home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/scheduler.go:125 +0xb9
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).SetPriorityID(0xc002da8f70, 0xb)
        /home/tobias/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/scheduler.go:240 +0x58
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).setDescLockedRaftMuLocked(0xc005d52a80, {0x506a918, 0xc0076fe8a0}, 0xc0033d2490)
```

Guess I'll play the bisect game later.

server.TestAdminAPITableStats [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4617668&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4617668&tab=artifacts#/) on master @ [d84f07e856dbb3d62f0336f4e8439280e2a018ed](https://github.com/cockroachdb/cockroach/commits/d84f07e856dbb3d62f0336f4e8439280e2a018ed):


```
=== RUN   TestAdminAPITableStats
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAdminAPITableStats3944828393
    test_log_scope.go:80: use -show-logs to present logs inline
    admin_cluster_test.go:102: 
        INSERT INTO system.web_sessions ("hashedSecret", username, "expiresAt")
        VALUES($1, $2, $3)
        RETURNING id
        : unexpected multiple results
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestAdminAPITableStats3944828393
--- FAIL: TestAdminAPITableStats (33.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPITableStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/scheduler.go
+++ b/pkg/kv/kvserver/scheduler.go
@@ -14,6 +14,7 @@ import (
 	"container/list"
 	"context"
 	"fmt"
+	"runtime/debug"
 	"sync"
 
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
@@ -76,6 +77,7 @@ type rangeIDQueue struct {
 
 	// High priority.
 	priorityID     roachpb.RangeID
+	priorityStack  []byte // for debugging in case of assertion failure; see #75939
 	priorityQueued bool
 }
 
@@ -123,9 +125,10 @@ func (q *rangeIDQueue) Len() int {
 func (q *rangeIDQueue) SetPriorityID(id roachpb.RangeID) {
 	if q.priorityID != 0 && q.priorityID != id {
 		panic(fmt.Sprintf(
-			"priority range ID already set: old=%d, new=%d",
-			q.priorityID, id))
+			"priority range ID already set: old=%d, new=%d, first set at:\n\n%s",
+			q.priorityID, id, q.priorityStack))
 	}
+	q.priorityStack = debug.Stack()
 	q.priorityID = id
 }
 

--- a/pkg/kv/kvserver/scheduler.go
+++ b/pkg/kv/kvserver/scheduler.go
@@ -121,10 +121,7 @@ func (q *rangeIDQueue) Len() int {
 }
 
 func (q *rangeIDQueue) SetPriorityID(id roachpb.RangeID) {
-	if q.priorityID != 0 && q.priorityID != id &&
-		// This assertion is temporarily disabled, see:
-		// https://github.com/cockroachdb/cockroach/issues/75939
-		false {
+	if q.priorityID != 0 && q.priorityID != id {
 		panic(fmt.Sprintf(
 			"priority range ID already set: old=%d, new=%d",
 			q.priorityID, id))

